sdv/constraints/tabular.py

```diff
@@ -401,3 +401,70 @@ def reverse_transform(self, table_data):
         table_data[self._column] = self._formula(table_data)
 
         return table_data
+
+
+class Rounding(Constraint):
+    """Round a column based on the specified number of digits.
+
+    Args:
+        columns (str or list[str]):
+            Name of the column(s) to round.
+        digits (int):
+            How much to round each column. All columns will be rounded to this
+            number of digits.
+        handling_strategy (str):
+            How this Constraint should be handled, which can be ``transform``
+            or ``reject_sampling``. Defaults to ``transform``.
+        tolerance (int):
+            When reject sampling, the sample data must be within this distance
+            of the desired rounded values.
+    """
+
+    def __init__(self, columns, digits, handling_strategy='transform', tolerance=None):
+        if digits > 15:
+            raise ValueError('The value of digits cannot exceed 15')
+
+        if tolerance is not None and tolerance >= 10**(-1 * digits):
+            raise ValueError('Tolerance must be less than the rounding level')
+
+        if isinstance(columns, str):
+            self._columns = [columns]
+        else:
+            self._columns = columns
+
+        self._digits = digits
+        self._round_config = {column: self._digits for column in self._columns}
```

Comment:
I probably missed something, but is there a reason we don't let users provide the config themselves? It seems like it wouldn't be a big stretch since we already create a map of column names to a digits value

Reply:
I believe it seemed overly complicated, because users could just specify multiple Rounding constraints if they wanted different columns to be rounded to different digits. (see https://github.com/sdv-dev/SDV/pull/484#pullrequestreview-695337713)

Reply:
Yeah but I believe that was under the assumption each column would have it's own constraint. If the constraint is implemented in such a way that it already checks all columns for the amount to round them then I think it would be nice for users to be able to specify that instead of having to make however many constraints

Reply:
I think the comment was referring to - one constraint specifies one rounding level, so if you want to implement multiple rounding levels (aka digits) you can specify multiple Rounding constraints. I think I like that abstraction better, rather than extending it to all possible rounding levels in the dataset.

Reply:
> I think the comment was referring to - one constraint specifies one rounding level, so if you want to implement multiple rounding levels (aka digits) you can specify multiple Rounding constraints. I think I like that abstraction better, rather than extending it to all possible rounding levels in the dataset.

That is right. The goal was to not overcomplicate both the implementation and the user interface.

Reply:
I see, but I don't think it would change the implementation or UI at all. You would just change fit to not create the dict if it's provided and everything else stays the same